llvm/tools/sycl-post-link/ModuleSplitter.h

```diff
@@ -36,19 +38,84 @@ using EntryPointVec = std::vector<const Function *>;
 
 struct EntryPointGroup {
   StringRef GroupId;
+
+public:
   EntryPointVec Functions;
+
+  EntryPointGroup(StringRef GroupId = "") : GroupId(GroupId) {}
+  EntryPointGroup(StringRef GroupId, EntryPointVec &&Functions)
+      : GroupId(GroupId), Functions(std::move(Functions)) {}
+
+  const StringRef getId() const { return GroupId; }
+  bool isEsimd() const;
+  bool isSycl() const;
+  void saveNames(std::vector<std::string> &Dest) const;
+  void rebuildFromNames(const std::vector<std::string> &Names, const Module &M);
+  void rebuild(const Module &M);
 };
 
 using EntryPointGroupVec = std::vector<EntryPointGroup>;
 
-struct ModuleDesc {
+enum class ESIMDStatus { SYCL_ONLY, ESIMD_ONLY, SYCL_AND_ESIMD };
```

Comment:
Minor comment: IMO, the name of this enum is a bit confusing. Wouldn't it be more clear to name it something like SplitModuleStatus or ModuleStatus?

Reply:
Agree. I'm thinking about something like `SyclEsimdSplitStatus`